### PR TITLE
fix(featurestate-permissions): Add misc extra checks

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Add misc checks to feature state view set

## How did you test this code?

Adds unit test cases

### CODE REVIEW

File: api/features/views.py [hunk 1]
```diff
@@ -394,10 +374,6 @@ def create(self, request, *args, **kwargs):
                 status=status.HTTP_201_CREATED,
                 headers=headers,
             )
-        else:
-            logger.error(serializer.errors)
```
Comment (gagantrivedi) [inline api/features/views.py:398]: We are changing the response structure for 400 here though, thoughts @matthewelwell

Comment (matthewelwell) [reply api/features/views.py:398]: I think it's fine 👍

File: api/environments/tests/test_views.py [hunk 2]
```diff
@@ -83,32 +83,6 @@ def test_should_return_identities_for_an_environment(self):
...
-            data=self.fs_put_template % (feature_state.id, True, "This is a value"),
-            content_type="application/json",
-        )
-
-        # Then
-        feature_state.refresh_from_db()
-
-        assert response.status_code == status.HTTP_200_OK
-        assert feature_state.get_feature_state_value() == "This is a value"
-        assert feature_state.enabled
```
Comment (matthewelwell) [inline api/environments/tests/test_views.py:110]: I understand removing the test from here, but I can't see that we've reimplemented this test anywhere?

File: api/features/serializers.py [hunk 3]
```diff
@@ -308,11 +308,33 @@ def save(self, **kwargs):
...
+            raise serializers.ValidationError(
+                "Cannot change the environment of a feature state"
+            )
+        return environment
+
     def validate(self, attrs):
-        environment = attrs.get("environment")
+        environment = (
+            attrs.get("environment")
+            or self.context["view"].get_environment_from_request()
```
Comment (matthewelwell) [inline api/features/serializers.py]: Should we just add the environment to the context instead?

File: api/environments/tests/test_views.py (hunk 2 above)
Comment (gagantrivedi) [reply api/environments/tests/test_views.py:110]: It's implemented here: https://github.com/Flagsmith/flagsmith/blob/main/api/tests/integration/features/featurestate/test_environment_featurestate_viewset.py

File: api/features/serializers.py (hunk 3 above)
Comment (gagantrivedi) [reply api/features/serializers.py]: That will trigger an extra query for actions that don't need this, list, for example

Comment (matthewelwell) [reply api/features/serializers.py]: Hmm... but it's a trivial query that would simplify the code. Alternatively, we could just check the action in the `get_serializer_context` ? Maybe this approach is best, but it does feel slightly weird calling methods on the view at this stage of the request.

Comment (gagantrivedi) [reply api/features/serializers.py]: Yeah, that's a good idea. have updated it

File: api/tests/unit/features/test_unit_features_views.py [hunk 4]
```diff
@@ -864,3 +864,158 @@ def test_create_feature_reaching_max_limit(client, project, settings):
...
+    # Then
+    assert response.status_code == status.HTTP_201_CREATED
+    response.json()["feature_state_value"] == new_value
+
+
+@pytest.mark.parametrize(
+    "client",
+    [(lazy_fixture("admin_master_api_key_client")), (lazy_fixture("admin_client"))],
+)
+def test_cannon_create_feature_state_for_feature_from_different_project(
```
Comment (matthewelwell) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
def test_cannot_create_feature_state_for_feature_from_different_project(
```

File: api/features/views.py [hunk 5]
```diff
@@ -341,46 +347,26 @@ def get_environment_from_request(self):
...
         DEPRECATED: please use `/features/featurestates/` instead.
         Override create method to add environment and identity (if present) from URL parameters.
         """
         data = request.data
-        environment = self.get_environment_from_request()
-        if (
-            environment.project.organisation
-            not in self.request.user.organisations.all()
-        ):
-            return Response(status.HTTP_403_FORBIDDEN)
```
Comment (matthewelwell) [inline api/features/views.py:362]: I guess this is handled by the permission checks?

Comment (gagantrivedi) [reply api/features/views.py:362]: Yep, correct

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**